### DISCUSSION TITLE
SEB does not overwrite macOS spellchecker

### DISCUSSION THREAD

**Discussion by cnotz** [SafeExamBrowser/seb-mac#387] [Q&A]
Even when spellchecking is not active in SEB (through right-clicking), it still checks spelling if not deactivated in macOS settings because SEB does not overwrite this system-wide setting. Is there any known workaround, especially for exams with BYOD?
Best Claudio

**Comment by danschlet** [comment]
Can you please add more details to your bug report, please create an issue and answer all questions in the template here: https://github.com/SafeExamBrowser/seb-mac/issues/new/choose

This definitely was working correctly when last tested it. So it's important to know what macOS version you used, which SEB version, which exam settings (provide login so we can reproduce the issue).

**Reply by cnotz** [reply]
https://github.com/SafeExamBrowser/seb-mac/issues/388

**Comment by cnotz** [comment]
The problem exists with inline-text-corrections that appear if you write in combination with classtime.com. This problem is seen on various macos and seb versions.

**Comment by danschlet** [comment]
This is very annoying, as it seems to happen sometimes (once I could observe it as well when I tested now), but often blocking the spell checker works as expected. Do you observe that it's never blocked in Classtime?

**Reply by cnotz** [reply]
No I observe it the same way as you do. It seems to depend on typing speed. But shouldn't it be blocked anyway?

**Reply by cnotz** [reply]
Is there still something to do for me or have you taken over this issue?

**Reply by danschlet** [reply]
If we would find a reliable way how to reproduce the issue that would help, but it seems impossible. For me it mostly works. I know that isn't good enough if you do language tests...

I will work on this as soon as time permits. There are too many issues open right now. Unfortunately Apple also doesn't seem to have a solution, as it even isn't blocked in AAC in macOS.

**Comment by danschlet** [comment]
In SEB 3.3.3, I blocked spell checking completely. Unfortunately the only fully reliable method to disable automatic spelling correction only works when applied directly after the app has started and cannot be changed later when SEB would receive some exam configuration and would reconfigure other settings to that config.

I will file a feature request at Apple to provide an API to disable/enable spell checking/correction dynamically. Such an API method is available in the legacy (classic) WebView, but not the modern browser engine (WKWebView).
